plugins/modules/system/filesystem.py

```diff
@@ -155,39 +168,43 @@ def fstype(self):
         return type(self).__name__
 
     def get_fs_size(self, dev):
-        """ Return size in bytes of filesystem on device. Returns int """
+        """Return size in bytes of filesystem on device (integer).
+           Should query the info with a per-fstype command that can access the
+           device whenever it is mounted or not, and parse the command output.
+           Parser must ensure to return an integer, or raise a ValueError.
+        """
         raise NotImplementedError()
 
     def create(self, opts, dev):
         if self.module.check_mode:
             return
 
         mkfs = self.module.get_bin_path(self.MKFS, required=True)
-        if opts is None:
-            cmd = "%s %s '%s'" % (mkfs, self.MKFS_FORCE_FLAGS, dev)
-        else:
-            cmd = "%s %s %s '%s'" % (mkfs, self.MKFS_FORCE_FLAGS, opts, dev)
+        cmd = [mkfs] + self.MKFS_FORCE_FLAGS.split()
+        if opts is not None:
+            cmd += opts.split()
```

Comment:
I think it would be better to make `opts` a list in the first place.

Reply:
What do you mean by *in the first place* ? Would it be better to use type list for this param, or would it be better to convert it into a list as soon as possible (right after module argument_spec, in main()) ?

Reply:
I would make it type list and never write it down as a string, but right away as a list everywhere it's used.

Reply:
`opts` is a module parameter, and is `string`. I converted it to list as soon as possible, right after module's argument_spec.